server/list.js

```diff
@@ -15,14 +15,16 @@ const driveType = process.env.DRIVE_TYPE
 const driveId = process.env.DRIVE_ID
 
 let currentTree = null // current route data by slug
+let currentFilenames = null // current list of filenames for typeahead
 let docsInfo = {} // doc info by id
 let tags = {} // tags to doc id
 let driveBranches = {} // map of id to nodes
 const playlistInfo = {} // playlist info by id
 
 // normally return the cached tree data
 // if it does not exist yet, return a promise for the new tree
-exports.getTree = () => currentTree || updateTree()
+exports.getTree = () => currentTree || updateTree().tree
```

Comment:
@afischer I know we discussed this in person, but let me know when this PR is ready for another look. Generally it seems right to me, but want to give it a more thorough read after we resolve the issue with accessing a property of an unresolved promise here.